server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/LocalStorageManager.scala

```diff
@@ -61,36 +64,35 @@ private[worker] final class DiskFlusher(
   private var lastBeginFlushTime: Long = -1
   def getLastFlushTime: Long = lastBeginFlushTime
 
-  @volatile
-  var stopFlag = false
+  var stopFlag = new AtomicBoolean(false)
 
   private val worker = new Thread(s"$this") {
     override def run(): Unit = {
-      while (!stopFlag) {
+      while (!stopFlag.get()) {
         val task = workingQueue.take()
-
-        val key = s"DiskFlusher-$workingDir"
-        workerSource.sample(WorkerSource.FlushDataTime, key) {
-          if (!task.notifier.hasException) {
-            try {
-              lastBeginFlushTime = System.nanoTime()
-              task.fileChannel.write(task.buffer.nioBuffers())
-            } catch {
-              // InterruptedIOException when notifyError
-              case _: ClosedByInterruptException =>
-              case e: IOException =>
-                task.notifier.setException(e)
-                stopFlag = true
-                logError(s"$this write failed, report to DeviceMonitor, exeption: $e")
-                reportError(workingDir, e, DeviceErrorType.ReadOrWriteFailure)
+        writeActionPool.submit(new Runnable {
+          override def run(): Unit = {
+            val key = s"DiskFlusher-$workingDir"
+            workerSource.sample(WorkerSource.FlushDataTime, key) {
```

Comment:
workerSource has an assumption of sequential update. This will break the behavior.